### PR TITLE
Marshall out iterables as lists in CPython

### PR DESCRIPTION
**Pull request by mmisol**
### Purpose

Adds support for marshalling out iterables that are not lists or
dictionaries using the CPython engine. In order to achieve this, the
check PyIter.IsIterable was used, along with some extra considerations,
mentioned in the source with comments.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner 

### FYIs

@DynamoDS/dynamo

### CODE REVIEW

File: test/Libraries/DynamoPythonTests/PythonEvalTests.cs [hunk 1]
```diff
@@ -195,5 +195,34 @@ class myobj:
                 Assert.AreEqual("Output could not be converted to a .NET value", exc.Message);
             }
         }
+
+        [Test]
+        public void NonListIterablesCanBeOutput()
```
Comment (mjkkirschner) [inline test/Libraries/DynamoPythonTests/PythonEvalTests.cs:200]: does this test hit the condition on line `204`?

Comment (mmisol) [reply test/Libraries/DynamoPythonTests/PythonEvalTests.cs:200]: Not this test. One which failed without that part is this one https://github.com/DynamoDS/Dynamo/blob/master/test/Libraries/DynamoPythonTests/PythonEvalTestsWithLibraries.cs#L189

Comment (mjkkirschner) [reply test/Libraries/DynamoPythonTests/PythonEvalTests.cs:200]: so it's currently failing?

Comment (mmisol) [reply test/Libraries/DynamoPythonTests/PythonEvalTests.cs:200]: No, sorry if I didn't say it right. That test does work and hits that line.

What I meant is that at some point during my development of this I noticed the test failed, which triggered the changes now in line 204.

**Comment by mmisol** [comment]
Tests passed here: https://master-15.jenkins.autodesk.com/view/DYN/job/DYN-DevCI_Self_Service/707/. Merging

**Pull request merged**